File: app/src/main/java/com/example/newsfeed/NewsApp/NewsActivity.java
package com.example.newsfeed.NewsApp;

import androidx.appcompat.app.AppCompatActivity;
import androidx.navigation.NavController;
import androidx.navigation.Navigation;
import androidx.navigation.ui.NavigationUI;

import android.os.Bundle;

import com.example.newsfeed.NewsApp.UI.NewsViewModel;
import com.example.newsfeed.R;
import com.google.android.material.bottomnavigation.BottomNavigationView;

public class NewsActivity extends AppCompatActivity {
    public NewsViewModel ViewModel;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_news);

    //    NewsRepository newsRepository = new NewsRepository(ArticlesDatabase.getInstance(this));
  //      NewsViewModelProviderFactory ViewModelProvideFactory=new NewsViewModelProviderFactory(newsRepository);
//ViewModel= new ViewModelProvider(this,ViewModelProvideFactory).get(NewsViewModel.class);

        BottomNavigationView bottomNavigationView=findViewById(R.id.bottomNavigationView);
        NavController navController= Navigation.findNavController(this,R.id.newsNavHostFragment);
        NavigationUI.setupWithNavController(bottomNavigationView,navController);
    }
}

File: app/src/main/java/com/example/newsfeed/NewsApp/NewsActivity.kt
package com.example.newsfeed.NewsApp

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.lifecycle.ViewModelProvider
import androidx.navigation.fragment.findNavController
import androidx.navigation.ui.setupWithNavController
import com.example.newsfeed.NewsApp.Database.ArticlesDatabase
import com.example.newsfeed.NewsApp.Repository.NewsRepository
import com.example.newsfeed.NewsApp.UI.NewsViewModel
import com.example.newsfeed.NewsApp.UI.NewsViewModelProviderFactory
import com.example.newsfeed.R
import kotlinx.android.synthetic.main.activity_news.*

class NewsActivity : AppCompatActivity() {

    lateinit var viewModel: NewsViewModel
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setTheme(R.style.Theme_NewsFeed)
        setContentView(R.layout.activity_news)

        //here we actually make declarations of newsViewModel and NewsRepository and NewsViewModelProviderFactory
        val newsRepository=NewsRepository(ArticlesDatabase.invoke(this))
        val newsViewModelProviderFactory=NewsViewModelProviderFactory(application,newsRepository)
        viewModel=ViewModelProvider(this, newsViewModelProviderFactory).get(NewsViewModel::class.java)

        // Here We actually set up bottom Navigation View For switching from one fragment to another
        bottomNavigationView.setupWithNavController(newsNavHostFragment.findNavController())


    }
}